core/src/Text/Interpolation/Nyan/Core/Internal/Parser.hs

```diff
@@ -275,14 +275,11 @@ switchesHelpMessage sopts =
       , val /= defVal
       ]
 
-intPieceP :: Ord e => Parsec e Text [ParsedIntPiece]
-intPieceP = asum
-  [
+intPieceP :: Ord e => SwitchesOptions -> Parsec e Text [ParsedIntPiece]
+intPieceP SwitchesOptions{..} = asum [
 
-    -- consume comments
-    string "--" >>= \prefix -> do
-      content <- takeWhile1P Nothing (/= '\n')
-      pure $ one $ PipComments (prefix <> content)
+    -- ignore comments if 'commenting' switch is on
+    guard commenting *> skipLineComment "--" $> []
```

Comment:
Yup, this way it looks simpler a lot. There's actually no reason to keep the comments for further processing, that was a bit of surprise )

Reply:
Looks good, thanks :star: